newsroom/section_filters/views.py

```diff
@@ -1,86 +1,108 @@
 import re
+import json
+from typing import Any, Dict, Optional
 
-from bson import ObjectId
-from quart_babel import gettext
+from quart import abort
+from pydantic import BaseModel, field_validator
 
-from superdesk.core import get_current_app
-from superdesk.flask import jsonify, request
-from superdesk import get_resource_service
+from superdesk.core.web import EndpointGroup, Response, Request
 
 from newsroom.decorator import admin_only
-from newsroom.section_filters import blueprint
-from newsroom.utils import (
-    get_json_or_400,
-    get_entity_or_404,
-    set_original_creator,
-    set_version_creator,
-)
-from newsroom.utils import query_resource
+from newsroom.core import get_current_wsgi_app
+from newsroom.utils import get_json_or_400
 
+from .service import SectionFiltersService
+from .model import SectionFilter
 
-def get_settings_data():
+section_filters_endpoints = EndpointGroup("section_filters", __name__)
+
+
+async def get_settings_data():
     """Get the settings data for section filter
 
     :param context
     """
+    all_filters = [obj async for obj in SectionFiltersService().get_all_raw()]
+
     data = {
-        "section_filters": list(query_resource("section_filters")),
-        "sections": get_current_app().as_any().sections,
+        "section_filters": all_filters,
+        "sections": get_current_wsgi_app().sections,
     }
+
     return data
 
 
-@blueprint.route("/section_filters", methods=["GET"])
+async def get_section_filter_or_abort(id: str) -> SectionFilter:
+    section_filter = await SectionFiltersService().find_by_id(id)
+    if section_filter is None:
+        abort(404)
```

Comment:
It would be good not use use Flask/Quart functionality directly, but rather through the newer `Request` class

Reply:
do you mean like passing down the request instance coming into the view function?

Reply:
done.